### DISCUSSION TITLE
Modbus TCP server alternative

### DISCUSSION THREAD

**Discussion by Ordepsousa** [TypQxQ/Sigenergy-Home-Assistant-Integration#21] [Show and tell]
Hi all,

Last month i developed myself an HA Sigenergy integration via API. But with fewer sensors, just the basic ones for PV, battery, grid import/export. But i was trying to find a local way because when there is a power/internet outage the service is down.
I didn't know that there was a Modbus TCP Server since I am a regular user and I can't see that option available.
Now I know there are alternatives. Awesome project!

And I'm writing because I found another way to run your integration without asking for Installer to modify options on Sigenergy APP. Because asking for something that you are not allowed to by default could be a nightmare to accomplish.

So, another way to do it is by having a converter from RS485 to RJ45 (ethernet) and wire it accordingly to the official manual of the inverter (page 14) pins 15 and 16
Manual inverter: https://b2b.aprilice.com/se_sv/mpattachment/file/download/id/612/
Example of RS485 to RJ45 device: https://www.waveshare.com/rs485-to-eth-b.htm

This way you have Modbus connection without having to ask to the installer to configure it. At least for me it worked out of the box.

**Comment by TypQxQ** [comment]
That is verry good to know as an alternative if you have a strange installer.
I personally would rather enable it through existing cables than have someone attach strange devices to the inverter.

P.S. I am also just an end-user 😄

**Comment by paddyflah84** [comment]
Hi, I am just getting into home assistant and I have a Segenenergy PV system. I was looking at your docs but I dont see an option for Modbus IP in my app. Is this something visible only to the installer that put in the system?

**Reply by TypQxQ** [reply]
Hi,
The IP address can be found inside your router configuration. I don”t know if it’s visible to the installer either. Typically, the IP address would be assigned by your local network's DHCP server (router) when the device connects to your network. You can check the IP address through your router's connected devices list

**Comment by paddyflah84** [comment]
Cheers. The option to enable modbus does not seem to be on my client app. I
presume that has to be enabled by my installer?

**Reply by TypQxQ** [reply]
Yes

**Comment by paddyflah84** [comment]
Thanks for the pointers. Looking forward to using the extension! Thanks a
million.

**Comment by paddyflah84** [comment]
Hi, I got in touch with my installer and he enabled the server port. I was able to see the modbus server and get the IP.

I implemented the config changes, but I seem to be having an issue with the modbus_sigenenergy.yaml file. Its returning exceptions in HA. Its complaining about syntax but no alternatives have been successful.

I created the packages directory, created the file and pasted the contents in. The secrets and config file changes are also in place.

I can see the sensors in the dashboard but they are all returning 0...

Sorry I'm a bit of a noob on HA...

Any thoughts?

[cid:7fe00b46-20eb-49f1-98c2-dc9d39f16914]

[cid:188e50d2-3cbe-48f2-9b6d-b5a61acb7d57]

As a side question, do I also need to change my device Operation Mode to use Remote EMS Mode?
Thanks

Patrick

________________________________

**Comment by paddyflah84** [comment]
I presume this is because Im running 2024.11...i see that in the docs now it only supports 2023.10. Is that the case, if it is indeed apologies 🙂

________________________________

**Reply by goatie999** [reply]
I’m running (just installed following the instructions this week) HA 2024.9.3 and that seems to be working fine for me.   Can’t assist with much else but can at least confirm that the installation works with the HA release from Sept this year.

**Reply by TypQxQ** [reply]
Works fine with 2024.11 too.

**Reply by paddyflah84** [reply]
Thanks very much. I reimaged my device and on a second go it started working. Thanks for all the pointers. Very cool stuff

**Comment by TypQxQ** [comment]
I'm going to close this because the current comments have nothing to do with original post.
Please open a new thread if it is not related to OP.
